### PR TITLE
Fix/accordion tabbar navigation height

### PR DESCRIPTION
**Pull request by simonmitchell**
## Description

Fixes the height of the accordion navigation bar to match the RHS view controller in the split view

## Motivation and Context

It looks nicer!

## How Has This Been Tested?

Tested locally in Nordstrom

## Screenshots (if appropriate):
Simulator Screen Shot - iPad Pro (12 9-inch) (3rd generation) - 2020-03-09 at 15 14 38

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ThunderCloud/AccordionTabBarItemTableViewCell.swift [hunk 1]
```diff
@@ -18,8 +18,10 @@ class AccordionTabBarItemTableViewCell: TableViewCell {
 	@IBOutlet weak var customTitleHeightConstraint: NSLayoutConstraint!
 	
 	@IBOutlet weak var viewControllerView: UIView!
-	
-	override func prepareForReuse() {
+	    
+    @IBOutlet weak var headerHeightConstraint: NSLayoutConstraint!
+    
+    override func prepareForReuse() {
```
Comment (simonmitchell) [inline ThunderCloud/AccordionTabBarItemTableViewCell.swift]: Will re-indent this file once approved :)

File: ThunderCloud/AccordionTabBarViewController.swift [hunk 2]
```diff
@@ -55,24 +55,26 @@ public class AccordionTabBarItem: Row {
...
     
     public var expanded: Bool = false
     
     public var isFirstItem: Bool = false
     
     public func height(constrainedTo size: CGSize, in tableView: UITableView) -> CGFloat? {
-        let height = expanded ? remainingHeight + 44 : 44
-        return isFirstItem ? height + 20 : height
+        let height = expanded ? remainingHeight + navigationBarHeight : navigationBarHeight
+        return isFirstItem ? height + UIApplication.shared.statusBarFrame.height : height
```
Comment (BenCShutt) [inline ThunderCloud/AccordionTabBarViewController.swift]: I know the `statusBarFrame` is fetched different when using iOS 13 `scenes` but guess this is OK if we don't use that?

File: ThunderCloud/AccordionTabBarViewController.swift [hunk 3]
```diff
@@ -311,14 +312,26 @@ open class AccordionTabBarViewController: TableViewController, StormObjectProtoc
         navigationController.view.sendSubviewToBack(navigationController.navigationBar)
     }
     
+    private var navigationBarHeight: CGFloat {
```
Comment (BenCShutt) [inline ThunderCloud/AccordionTabBarViewController.swift]: Can we not use safe areas? Constraining to `superview` vs `safeArea` where required?

File: ThunderCloud/AccordionTabBarViewController.swift (hunk 2 above)
Comment (simonmitchell) [reply ThunderCloud/AccordionTabBarViewController.swift]: Yeah don't think we'd have to worry too much about that for now :) A lot of Storm will need a re-think once we start using SwiftUI anyways :D

File: ThunderCloud/AccordionTabBarViewController.swift [hunk 4]
```diff
@@ -101,7 +101,7 @@ public class AccordionTabBarItem: Row {
         accordionCell.cellTextLabel?.textColor = contrastingColor
         accordionCell.cellImageView?.tintColor = contrastingColor
         
-        accordionCell.topConstraint.constant = isFirstItem ? 28 : 8
+        accordionCell.topConstraint.constant = isFirstItem ? UIApplication.shared.statusBarFrame.height : 0
```
Comment (BenCShutt) [inline ThunderCloud/AccordionTabBarViewController.swift]: Not sure why `28` was used here and `20` above, do we need `8` padding?

**Comment by BenCShutt** [comment]
Changes look good to me :), but would be interested to hear your response about constraining to `safeArea` and `superview` instead of using explicit sizes

File: ThunderCloud/AccordionTabBarViewController.swift (hunk 3 above)
Comment (simonmitchell) [reply ThunderCloud/AccordionTabBarViewController.swift]: Unsure if safe areas will satisfy our requirements... This is used to provide the height of one particular view within a stack view within the cells that make up the left hand panel of the accordion view. The navigation bar is completely hidden, so wouldn't really want to rely on safe areas (And wouldn't be sure exactly where to access it from e.t.c)

File: ThunderCloud/AccordionTabBarViewController.swift (hunk 4 above)
Comment (simonmitchell) [reply ThunderCloud/AccordionTabBarViewController.swift]: Previously this was the spacing between the top of the cell and the image/text (Kinda like padding) now it's more like margin, and the 8 points of vertical padding isn't really needed any more as we rely on aligning centre vertical with the parent (And also > constants of 0)

File: ThunderCloud/AccordionTabBarViewController.swift (hunk 3 above)
Comment (BenCShutt) [reply ThunderCloud/AccordionTabBarViewController.swift]: Sure :) Let's go with this then! :P

**Pull request merged**